File: core/src/main/java/inetsoft/web/binding/event/SetCellBindingEvent.java
/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.web.binding.event;

import inetsoft.web.binding.model.table.CellBindingInfo;
import inetsoft.web.binding.model.table.TableCell;

/**
 * Class that encapsulates the parameters for applying a selection.
 *
 * @since 12.3
 */
public class SetCellBindingEvent {
   /**
    * Get the assembly name.
    * @return assembly name.
    */
   public String getName() {
      return name;
   }

   /**
    * Set the assembly name.
    * @param name assembly name.
    */
   public void setName(String name) {
      this.name = name;
   }

   /**
    * Get the select cells.
    * @return select cells.
    */
   public TableCell[] getSelectCells() {
      return selectCells;
   }

   /**
    * Set the select cells.
    * @param selectCells select cells.
    */
   public void setSelectCells(TableCell[] selectCells) {
      this.selectCells = selectCells;
   }
   
   /**
    * Get the cell binding.
    * @return cell binding.
    */
   public CellBindingInfo getBinding() {
      return binding;
   }

   /**
    * Set the cell binding.
    * @param binding the cell binding.
    */
   public void setBinding(CellBindingInfo binding) {
      this.binding = binding;
   }

   private String name;
   private TableCell[] selectCells;
   private CellBindingInfo binding;
}

File: web/projects/portal/src/app/binding/event/set-cell-binding-event.ts
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../../common/viewsheet-client/index";
import { CalcTableCell } from "../../common/data/tablelayout/calc-table-cell";
import { CellBindingInfo } from "../data/table/cell-binding-info";

/**
 * Event for common parameters for composer object events.
 */
export class SetCellBindingEvent implements ViewsheetEvent {
   /**
    * Creates a new instance of <tt>GetTableModelEvent</tt>.
    *
    * @param chartName the name of the viewsheet object.
    */
   constructor(name: string, selectCells: CalcTableCell[], binding: CellBindingInfo) {
      this.name = name;
      this.selectCells = selectCells;
      this.binding = binding;
   }

   private name: string;
   private selectCells: CalcTableCell[];
   private binding: CellBindingInfo;
}
